core/src/test/scala/neotypes/CleaningIntegrationSpec.scala

```diff
@@ -7,13 +7,14 @@ import scala.concurrent.{Await, Future}
 import scala.concurrent.duration.Duration
 
 /** Base class for integration specs that require to clean the graph after each test. */
-abstract class CleaningIntegrationSpec[F[_]](testkit: EffectTestkit[F]) extends BaseIntegrationSpec(testkit) {
+trait CleaningIntegrationSpec[F[_]] extends BaseIntegrationSpec[F] { self: SessionProvider[F] =>
   override final def withFixture(test: NoArgAsyncTest): FutureOutcome = {
-    complete {
-      super.withFixture(test)
-    } lastly {
-      this.cleanDb()
-    }
+    val r = for {
+      o <- super.withFixture(test).toFuture
+      _ <- this.cleanDB()
+    } yield o
+
+    new FutureOutcome(r)
```

Comment:
This has the consequence that if a test fails then the clean won't be executed which probably will result in more test fails, but I was unable to keep the previous blocking clean without breaking **Scalatest**.